### PR TITLE
Add JSONC language mode for .markdownlintrc files

### PR DESCRIPTION
**Pull request by thernstig**
Add JSONC language mode for .markdownlintrc files (markdownlint.json was not changed, see reasons in comments below).

### CODE REVIEW

**Comment by thernstig** [comment]
@DavidAnson Hope you like it :)

**Comment by DavidAnson** [comment]
By default, the [config file helpers](https://github.com/DavidAnson/markdownlint/blob/master/README.md#config) used by the core markdownlint library do not support JSONC, so I am reluctant to encourage it with a change like this.

That said, the RC format is used primarily with the CLI which does support JSONC, so I am less concerned about adding that here – especially because it doesn’t have a file extension that allows VS Code to guess the right format.

At the same time, I would prefer the JSON naming convention to be adopted and considered the “preferred“ name vs. the RC format, so I’m not sure about advertising JSONC for one and not the other.

I will think about this some more.

File: package.json [hunk 1]
```diff
@@ -56,6 +56,15 @@
 		"onCommand:markdownlint.fixLine"
 	],
 	"contributes": {
+		"languages": [
+			{
+				"id": "jsonc",
+				"filenames": [
+					".markdownlint.json",
```
Comment (DavidAnson) [inline package.json:63]: Let's please remove this so it's only for `.markdownlintrc` for now.

File: package.json [hunk 2]
```diff
@@ -56,6 +56,15 @@
 		"onCommand:markdownlint.fixLine"
 	],
 	"contributes": {
+		"languages": [
```
Comment (DavidAnson) [inline package.json:59]: Please move this section below "jsonValidation" to (mostly) sort the non-"configuration" items. ("menus" is out of place, but that's my fault.)

**Comment by thernstig** [comment]
All done. I did wish it worked for .markdownlint.json since that is what I was using myself, and now I am forced to add this to my own workspace VS Code config. Maybe you can fix it for the future so the default config file helpers default to jsonc.

**Comment by DavidAnson** [comment]
Thank you! The reason JSONC is not supported by default for the core library is that doing so would require adding an additional dependency and I try hard to limit the number of dependencies there to just the one that is the Markdown parser. The config API does allow the caller to pass in its own file handler and that’s how the CLI and this extension work with JSONC. However, that requires the caller to anticipate this situation and I’m afraid many will not. If you have any thoughts or suggestions here, I would love to find a solution that makes us both happy!

**Comment by thernstig** [comment]
From what I gather, I believe the only solution is if you added jsonc to the core lib, but I also completely get behind the idea of few dependencies. It's a pickle!

One option would be to add an optionalPeerDependencies maybe, and use jsonc if that is installed, but that might be a bad idea (or maybe I completely misunderstand how to use the optionalPeerDependencies).

Suppose I got no really good ideas unfortunately.

**Comment by DavidAnson** [comment]
Okay, let’s go with this for now. I should be able to merge the PR this afternoon. Thanks!

**Pull request merged**